configuration header (#112)
ComputationalRadiationPhysics/cupla: doc/StandaloneHeader.md

Standalone Header for *cupla*
=============================

> cupla provides stand alone header to use the library without CMake e.g in a cling environment.

Reviewer: ```suggestion
cupla provides stand alone header to use the library without CMake, e.g in a [cling](https://github.com/root-project/cling) environment.
```

Available Pre-compiler Definitions 
=================================

Pre-compiler definitions can be used to control the behavior of cupla/alpaka.

> The definitions must be passed via compiler instruction or defined before the accelerator header is included.

Reviewer: ```suggestion
The definitions must be passed via a compiler flag or be defined before the accelerator header is included.
```

> - `CUPLA_STREAM_ASYNC_ENABLED` 0 == use synchronous streams, 1 == use asynchronous streams

Reviewer: ```suggestion
- `CUPLA_STREAM_ASYNC_ENABLED`: `0` use synchronous streams, `1` use asynchronous streams
```

> - `CUPLA_HEADER_ONLY` 1 == *cupla* will be used as header only library (default), else you must compile all cpp files from [src](https://github.com/ComputationalRadiationPhysics/cupla/tree/master/src)

Reviewer: ```suggestion
- `CUPLA_HEADER_ONLY`: `1` *cupla* will be used as header-only library (default), otherwise you must compile all `.cpp` files in [`src/`](https://github.com/ComputationalRadiationPhysics/cupla/tree/master/src)
```

linker dependencies
===================

> Depending of the used accelerator you must link the library `pthread` or activate OpenMP support.

Reviewer: ```suggestion
Depending of the used accelerator you must link the library `pthread` and/or activate OpenMP support.
```

Example
=======

> To select an accelerator you must include the corresponding accelerator header from [cupla/standalone](https://github.com/ComputationalRadiationPhysics/cupla/tree/master/include/cupla/standalone)

Reviewer: ```suggestion
To select an accelerator you must include the corresponding accelerator header from [`cupla/standalone/`](https://github.com/ComputationalRadiationPhysics/cupla/tree/master/include/cupla/standalone)
```

```C++

// use synchronous streams for the accelerator
#define CUPLA_STREAM_ASYNC_ENABLED 0

> // use cupla as header only library

Reviewer: ```suggestion
// use cupla as header-only library
```
